src/autora/doc/runtime/prompts.py

```diff
@@ -1,17 +1,44 @@
+from __future__ import annotations
+
 from enum import Enum
 
 LLAMA2_INST_CLOSE = "[/INST]\n"
 
-# Standard Llama2 template
-TEMP_LLAMA2 = """
-[INST]<<SYS>>
-{sys}
 
-{instr}
+class PromptBuilder:
+    """
+    Utilty class for building LLAMA2 prompts. Uses a stateful builder pattern.
+    """
+
+    def __init__(self, sys: str, instr: str):
+        self.instr = instr
+        self.prompt_text = f"""<s>[INST] <<SYS>>
+{ sys }
+<</SYS>>"""
```

Comment:
These texts are really hard to read! I wonder if you can use [`textwrap.dedent`](https://docs.python.org/3/library/textwrap.html#textwrap.dedent) for these prompt texts?
```suggestion
        self.prompt_text = textwrap.dedent(f"""\
        <s>[INST] <<SYS>>
        { sys }
        <</SYS>>
        """)
```

Reply:
Good idea, although I tried `textwrap.dedent` and it was very finicky about every single line having the exact trailing characters so it didn't do the trick. But I add a one-liner function to do this, along with some comments for better clarity. But I guess some familiarity with the llama2 prompt format is expected.